### PR TITLE
Allow specifying dialect through parsers facade

### PR DESCRIPTION
**Pull request by gabro**
This PR allows specifying a dialect through the JS api.

Here's an example of usage from ASTExplorer:

This PR is based on #957 (which in turn is based on #956). I'll rebase as soon as the parent PRs get merged.

### CODE REVIEW

File: scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala [hunk 1]
```diff
@@ -56,22 +66,41 @@ object JSFacade {
...
+
+  private[this] def parse[A <: Tree: Parse](
+    code: String,
+    settings: Settings
+  ): js.Dictionary[Any] = {
+    val dialect = (for {
+      settings <- settings.toOption
+      dialectStr <- settings.get("dialect")
+      dialect <- Dialect.standards.get(dialectStr)
+    } yield dialect).getOrElse(dialects.Scala211)
```
Comment (olafurpg) [inline scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala]: Default to 2.12?

Comment (gabro) [inline scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala]: I wanted to keep it consistent with the current meta behavior. Is that going to change?

Comment (olafurpg) [inline scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala]: Good point, let's keep it then. The default dialect is planned to change soon, either to something lenient like https://github.com/scalameta/scalafmt/blob/d204ae1437247aae0f0b8a9901e433368057e678/scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala#L43 or maybe even go away for parsing/pretty-printing.

Comment (gabro) [inline scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala]: Cool, I'll be glad to drop the option completely in the future.

**Comment by gabro** [comment]
As per the rebase, I'd wait for #957 to get merged then I'll rebased on master.

**Comment by gabro** [comment]
@olafurpg rebased

File: scalameta/parsers/js/src/test/scala/scala/meta/parsers/JSFacadeSuite.scala [hunk 2]
```diff
@@ -193,5 +193,54 @@ class JSFacadeSuite extends FunSuite {
...
 
+  test("default dialect is Scala 2.11") {
+    val code =
+      """|List(
+         |  1,
+         |  2,
+         |)""".stripMargin
+    val parsedDefaultDialect = JSFacade.parseStat(code)
+    val expected = d(
+      "error" -> "illegal start of simple expression"
```
Comment (olafurpg) [inline scalameta/parsers/js/src/test/scala/scala/meta/parsers/JSFacadeSuite.scala]: I just noticed the position is not included, do you think it would be useful to have it? It can be a separate PR or part of this one.

Comment (gabro) [inline scalameta/parsers/js/src/test/scala/scala/meta/parsers/JSFacadeSuite.scala]: Yes, I've been thinking about it.
I've added it to this PR for simplicity.

**Comment by gabro** [comment]
woops, the latest commit broke some tests, it should be ok now

**Comment by olafurpg** [comment]
There seems to be an error in an unrelated JS parsers test

```scala
[info] - parse Lit.Double *** FAILED ***
[info]   "...2,
[info]     "syntax": "42.2[d]"
[info]   }" did not equal "...2,
[info]     "syntax": "42.2[]"
[info]   }" (JSFacadeSuite.scala:19)
```

**Comment by gabro** [comment]
Odd, it fails only with Scala 2.12. It seems like `syntax` acts differently across version of Scala for Double :O

**Comment by gabro** [comment]
I can reproduce this on my machine, by running `parsersJS/test` after setting the version to Scala 2.12.2

**Comment by olafurpg** [comment]
Interesting @gabro, the bug happens only in 2.12 JS. I opened https://github.com/scalameta/scalameta/issues/961, I propose we remove that test and I can follow up on that issue in a separate PR.

**Comment by gabro** [comment]
I agree, I've ignored the test and added a reference to #961

File: scalameta/parsers/js/src/test/scala/scala/meta/parsers/JSFacadeSuite.scala [hunk 3]
```diff
@@ -193,5 +195,56 @@ class JSFacadeSuite extends FunSuite {
...
+    check(parsedDefaultDialect, expected)
+  }
+
+  test("inexisting dialects fallback to Scala 2.11") {
+    val code =
+      """|List(
+         |  1,
+         |  2,
+         |)""".stripMargin
+    val parsedDefaultDialect = JSFacade.parseStat(code, js.Dictionary("dialect" -> "wrong"))
```
Comment (gabro) [inline scalameta/parsers/js/src/test/scala/scala/meta/parsers/JSFacadeSuite.scala]: Right, probably if you pass a wrong one you want to know. It makes sense I'll report an error (instead of throwing an exception)

**Comment by gabro** [comment]
@olafurpg done, take a look
